### DISCUSSION TITLE
PIX on Windows

### DISCUSSION THREAD

**Discussion by Syncaidius** [dotnet/Silk.NET#1904] [Q&A] [answered]
Apologies if this has already been asked or implemented, but I could not find any results on this topic, after searching the repo issues/discussions!

Are there any plans to generate bindings for PIX on Windows? 
https://devblogs.microsoft.com/pix/winpixeventruntime/

Without it, there doesn't seem to be a way to add debug events or markers to a D3D12 command list. `BeginEvent()`, `EndEvent()` and `SetMarker()` do exist on `ID3D12GraphicsCommandList`, but they only accept PIX data parameters.

In DX11 this functionality was previously part of the D3D11 debug layer API, so it wasn't a separate library.

**Comment by curin** [comment]
For 2.X we aren't adding any new bindings, and 3.0 is currently still in development focusing on getting to parity with the bindings currently available.

**Reply by Beyley** [reply, marked as answer]
I'll open a tracking issue, since this is definitely a usecase we would like to provide for 3.0

**Reply by Syncaidius** [reply]
Thanks both. That's really good to know!
